exonum-java-binding/core/src/main/proto/service.proto

```diff
@@ -0,0 +1,42 @@
+// Copyright 2020 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package exonum.binding;
+
+option java_package = "com.exonum.binding.common.messages";
+
+/**
+TODO: move this file to exonum-proto-sources-repo
+currently this action is blocked till native update to the latest revision
+*/
+
+// An object for the service configuration which passed to `service#initialize` method.
+// It brings a possibility to configure a service using any of provided formats.
+// The format provides the type of the configuration encoded as string value.
+message ServiceConfiguration {
+    enum Format {
+        NONE = 0;
+        // Identifies that the given configuration represented as `plain text`.
+        TEXT = 1;
+        // Identifies that the given configuration represented as Json.
+        JSON = 2;
+        // Identifies that the given configuration represented as Properties,
+        // i.e. `key=value` pairs.
```

Comment:
There is actually way more to this, I'd link

        // Identifies that the given configuration represented as Java Properties,
        // i.e., key value pairs.
        // See the format specification in j.u.Properties: https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Properties.html#load(java.io.Reader)